File: src/main/java/team404/utils/HttpClient.java
package team404.utils;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.bukkit.Bukkit;

import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class HttpClient {
    private static final ObjectMapper objectMapper = new ObjectMapper();

    public static void sendPostRequest(String urlString, Object request) {
        try {
            URL url = new URL(urlString);
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
            connection.setRequestMethod("POST");
            connection.setDoOutput(true);
            connection.setRequestProperty("Content-Type", "application/json");

            String requestBody = objectMapper.writeValueAsString(request);
            try (OutputStream os = connection.getOutputStream()) {
                byte[] input = requestBody.getBytes(StandardCharsets.UTF_8);
                os.write(input, 0, input.length);
            }

            int responseCode = connection.getResponseCode();
            if (responseCode != HttpURLConnection.HTTP_OK) {
                Bukkit.getLogger().warning("Request was not send, response code: %s".formatted(responseCode));
            }

            connection.disconnect();
        } catch (Exception e) {
            e.printStackTrace();
            Bukkit.getLogger().warning("Request was not send");
        }
    }
}

File: src/main/kotlin/team404/client/HttpClient.kt
package team404.client

import com.fasterxml.jackson.databind.ObjectMapper
import org.slf4j.LoggerFactory
import team404.constant.PLUGIN_NAMESPACE
import java.net.HttpURLConnection
import java.net.URI
import java.nio.charset.StandardCharsets

object HttpClient {
    private val objectMapper = ObjectMapper()
    private val logger = LoggerFactory.getLogger(PLUGIN_NAMESPACE)

    fun sendPostRequest(urlString: String, request: Any) {
        try {
            val url = URI(urlString).toURL()
            val connection = url.openConnection() as HttpURLConnection
            connection.requestMethod = "POST"
            connection.doOutput = true
            connection.setRequestProperty("Content-Type", "application/json")

            val requestBody = objectMapper.writeValueAsString(request)
            connection.outputStream.use { os ->
                val input = requestBody.toByteArray(StandardCharsets.UTF_8)
                os.write(input, 0, input.size)
            }

            val responseCode = connection.responseCode
            if (responseCode != HttpURLConnection.HTTP_OK) {
                logger.warn("Request was not sent, response code: $responseCode")
            }

            connection.disconnect()
        } catch (e: Exception) {
            logger.warn("Request was not sent", e)
        }
    }
}
